---
title: "Behavior and Electrophysiology Data Analysis"
author: "Rayna M Harris"
date: "January 14, 2017"
output:
  md_document:
    variant: markdown_github
---

```{r setup, echo = F}
# Part 1: Reading and analyzing beahvior and physiology data

## load libraries -----
library(dplyr) ## for filtering and selecting rows
library(plyr) ## for renmaing factors
library(ggplot2) ## for awesome plots!
library(reshape2) #@ for melting dataframe
library(ggdendro) ## for dendrograms!!
library(cowplot) ## for easy modification to themes

library(MCMC.qpcr) # for qpcr analysis
library(reshape) #for making data wide
library(corrplot) # simple corrplot with size-circles
library(PerformanceAnalytics) # fancy ass correlation plot
library(Hmisc) # for correlation stats

knitr::opts_chunk$set(fig.path = '../results/')
```

## Behavior

```{r figureoptions, echo = F}

## color palettes
WTPalette <- c('black','red')
FMFR1Palette <- c('black','red')

colvalAPA12 <- c("WT_untrained" = "black",  
                 "WT_trained" = "red", 
                 "FMR1-KO_untrained" = "black", 
                 "FMR1-KO_trained" = "red")

colvalAPA13 <- c("WT_control" = "black",  
                 "WT_trained" = "red", 
                 "FMR1-KO_control" = "black", 
                 "FMR1-KO_trained" = "red")

colorvalgenoAPA2 <-  c("WT_untrained" = "black" , 
                       "WT_trained" = "red",  
                       "FMR1-KO_untrained" = "white",
                       "FMR1-KO_trained" = "white") 

colorvalgenoAPA3 <-  c("WT_control" = "black" , 
                       "WT_trained" = "red",  
                       "FMR1-KO_control" = "white",
                       "FMR1-KO_trained" = "white") 


## jitter points
dodge <- position_dodge(.3)
```

```{r data wrangle, echo=F}

## wrangle the raw wt and fmr1 dataframes ----

## read the data 
wt <- read.csv("../data/01_WTbehavephys.csv", header=TRUE, stringsAsFactors = FALSE, na.strings = c("", "ND", "N/A"))
fmr1 <- read.csv("../data/01_FMR1behavephys.csv", header=TRUE, stringsAsFactors = FALSE, na.strings = c("", "ND", "N/A"))

## fixing columns both fmr1 and wt are as equivilent as possible
wt$filename <- as.character(paste(wt$filename, wt$X, sep="_")) ## mgerge 2 columns 

## delete obsolete columns
wt$X <- NULL 
wt$X.1 <- NULL
fmr1$X <- NULL
wt$next. <- NULL
fmr1$next. <- NULL
fmr1$PopSpike.max........mV. <- NA  ## add column
fmr1$PospSpike.Probability <- NA  ## add column
fmr1 <- fmr1[-c(75:80), ] ## removing rows
names(wt)
names(fmr1)


## create ind column with a animal name that matches qpcr (e.g "FMR1 AB" or "BL 19")
wt$ind <- wt$filename
head(wt$ind)
wt$ind <- gsub("[[:blank:]]*bl", "BL ", wt$ind) ##removed blank space then changes bl1... to BL 1...
wt$ind <- gsub("D[[:print:]]*", "", wt$ind) ## deletes the D (for Day) and everythign thereafter
wt$ind

fmr1$ind <- fmr1$filename
fmr1$ind <- gsub("RoomTrack_", "", fmr1$ind) ##remove RommTrack_"
fmr1$ind <- gsub("fmr1", "FMR1 ", fmr1$ind) ## changes to FMR1
fmr1$ind <- gsub("frm1", "FMR1 ", fmr1$ind) ## because one was typed wrong
fmr1$ind <- gsub("wildtype", "BL ", fmr1$ind) ## that one wildtype sample name
fmr1$ind <- gsub("_[ptr][[:print:]]*", "", fmr1$ind) ##deletes the rest of the filename
fmr1 <- filter(fmr1, !grepl("wildtypeA", filename ))  ## remove the 1 wildtype ind
fmr1$ind


## rename columns
names(wt)[3] <- "TotalTime"
names(wt)[4] <- "TotalPath"
names(wt)[5] <- "Entrances"
names(wt)[6] <- "TimeToFirstEntrance"
names(wt)[7] <- "PathToFirstEntrance"
names(wt)[8] <- "SpeedToFirstEntrance"
names(wt)[9] <- "EntrancePerDistance"
names(wt)[10] <- "TotalShocks"
names(wt)[11] <- "TimeToFirstShock"
names(wt)[12] <- "PathToFirstShock"
names(wt)[13] <- "Speed"
names(wt)[14] <- "SDofSpeed"
names(wt)[15] <- "LineArity"
names(wt)[16] <- "MaxAvoidTime"
names(wt)[17] <- "MaxAvoidPath"
names(wt)[18] <- "TimeToSecondEntrance"
names(wt)[19] <- "PathToSecondEntrance"
names(wt)[20] <- "SpeedToSecondEntranc"
names(wt)[21] <- "TimeTARG"
names(wt)[22] <- "pTimeTARG"
names(wt)[23] <- "pTimeCCW"
names(wt)[24] <- "pTimeOPP"
names(wt)[25] <- "pTimeCW"
names(wt)[26] <- "RayleigLength"
names(wt)[27] <- "RayleigAngle"
names(wt)[28] <- "PolarAveVal"
names(wt)[29] <- "PolarSdVal"
names(wt)[30] <- "PolarMinVal"
names(wt)[31] <- "PolarMinBin"
names(wt)[32] <- "MinLoBin"
names(wt)[33] <- "MinHiBin"
names(wt)[34] <- "PolarMaxVal"
names(wt)[35] <- "PolarMaxBin"
names(wt)[36] <- "MaxLoBin"
names(wt)[37] <- "MaxHiBin"
names(wt)[38] <- "AnnularMinVal"
names(wt)[39] <- "AnnularMinBin"
names(wt)[40] <- "AnnularMaxVal"
names(wt)[41] <- "AnnularMaxBin"
names(wt)[42] <- "AnnularAvg"
names(wt)[43] <- "AnnularSD"
names(wt)[44] <- "AnnularSkewnes"
names(wt)[45] <- "AnnularKurtosis"
names(wt)[46] <- "ind_bad"
names(wt)[47] <- "genotype_bad"
names(wt)[48] <- "APA"
names(wt)[49] <- "TotalLocomotorActivity"
names(wt)[50] <- "LastLocomotorActivity"
names(wt)[51] <- "TotalPunishment"
names(wt)[53] <- "LastPunishment"
names(wt)[54] <- "LastTotalEntrances"
names(wt)[55] <- "T1Retention"
names(wt)[56] <- "T2Retention"
names(wt)[57] <- "IO_Max"
names(wt)[58] <- "LTP_Baseline"
names(wt)[59] <- "LTP_Baseline_SD"
names(wt) # check all good

## do same for fmr1
names(fmr1)[3] <- "TotalTime"
names(fmr1)[4] <- "TotalPath"
names(fmr1)[5] <- "Entrances"
names(fmr1)[6] <- "TimeToFirstEntrance"
names(fmr1)[7] <- "PathToFirstEntrance"
names(fmr1)[8] <- "SpeedToFirstEntrance"
names(fmr1)[9] <- "EntrancePerDistance"
names(fmr1)[10] <- "TotalShocks"
names(fmr1)[11] <- "TimeToFirstShock"
names(fmr1)[12] <- "PathToFirstShock"
names(fmr1)[13] <- "Speed"
names(fmr1)[14] <- "SDofSpeed"
names(fmr1)[15] <- "LineArity"
names(fmr1)[16] <- "MaxAvoidTime"
names(fmr1)[17] <- "MaxAvoidPath"
names(fmr1)[18] <- "TimeToSecondEntrance"
names(fmr1)[19] <- "PathToSecondEntrance"
names(fmr1)[20] <- "SpeedToSecondEntranc"
names(fmr1)[21] <- "TimeTARG"
names(fmr1)[22] <- "pTimeTARG"
names(fmr1)[23] <- "pTimeCCW"
names(fmr1)[24] <- "pTimeOPP"
names(fmr1)[25] <- "pTimeCW"
names(fmr1)[26] <- "RayleigLength"
names(fmr1)[27] <- "RayleigAngle"
names(fmr1)[28] <- "PolarAveVal"
names(fmr1)[29] <- "PolarSdVal"
names(fmr1)[30] <- "PolarMinVal"
names(fmr1)[31] <- "PolarMinBin"
names(fmr1)[32] <- "MinLoBin"
names(fmr1)[33] <- "MinHiBin"
names(fmr1)[34] <- "PolarMaxVal"
names(fmr1)[35] <- "PolarMaxBin"
names(fmr1)[36] <- "MaxLoBin"
names(fmr1)[37] <- "MaxHiBin"
names(fmr1)[38] <- "AnnularMinVal"
names(fmr1)[39] <- "AnnularMinBin"
names(fmr1)[40] <- "AnnularMaxVal"
names(fmr1)[41] <- "AnnularMaxBin"
names(fmr1)[42] <- "AnnularAvg"
names(fmr1)[43] <- "AnnularSD"
names(fmr1)[44] <- "AnnularSkewnes"
names(fmr1)[45] <- "AnnularKurtosis"
names(fmr1)[46] <- "ind_bad"
names(fmr1)[47] <- "genotype_bad"
names(fmr1)[48] <- "APA"
names(fmr1)[49] <- "TotalLocomotorActivity"
names(fmr1)[50] <- "LastLocomotorActivity"
names(fmr1)[51] <- "TotalPunishment"
names(fmr1)[53] <- "LastPunishment"
names(fmr1)[54] <- "LastTotalEntrances"
names(fmr1)[55] <- "T1Retention"
names(fmr1)[56] <- "T2Retention"
names(fmr1)[57] <- "IO_Max"
names(fmr1)[58] <- "LTP_Baseline"
names(fmr1)[59] <- "LTP_Baseline_SD"

## add columns genotype (easier for wt df than for fmr1 df)
wt$genotype <- as.factor("WT")
fmr1$genotype <- ifelse(grepl("fmr1|frm1",fmr1$filename),'FMR1-KO','WT')
fmr1$genotype <- as.factor(fmr1$genotype)

## binding, clearning, filtering, renaming, adding conditions ----

wtfmr1 <- rbind(wt, fmr1) ## combine the datasets into one
wtfmr1 <- filter(wtfmr1, grepl("Room", filename)) ## remove non-data rows
str(wtfmr1)

## making strings factors
wtfmr1$ind <- as.factor(wtfmr1$ind)  
wtfmr1$APA <- as.factor(wtfmr1$APA) 

## making a bunch of columns numeric
wtfmr1[, c(3:45, 49:56, 58:59)] <- sapply(wtfmr1[, c(3:45, 49:56, 58:59)], as.numeric)
str(wtfmr1)

## add column session with APA information
wtfmr1$session <- ifelse(grepl("pretraining|pretrain|Hab", wtfmr1$filename), "pretraining", 
                     ifelse(grepl("training1|Train1", wtfmr1$filename), "training1",
                            ifelse(grepl("training2|Train2", wtfmr1$filename), "training2",
                                   ifelse(grepl("training3|Train3", wtfmr1$filename), "training3",
                                          ifelse(grepl("retention|reten", wtfmr1$filename), "retention", 
                                                 ifelse(grepl("Retest", wtfmr1$filename), "retest", "NA"))))))

wtfmr1$session  ## check that all names good with no NAs                                 
wtfmr1$session <- as.factor(wtfmr1$session)  
wtfmr1$session <- factor(wtfmr1$session, levels = c("pretraining", "training1", "training2", "training3", "retention", "retest"))

### separate out the summary columns and clean up / wrangle ----
summary <- filter(wtfmr1, session %in% c("retention", "retest"))
summary <- summary[c(60:61,48,62,49:59,1)] #select and reorder columns, so animal first
names(summary)
str(summary)

## rename APA to match qpcr data
#summary$APA <- revalue(summary$APA, c("untrained" = "control"))  # using "untrained" is more informative than control
summary$APA <- factor(summary$APA, levels = c("untrained", "trained"))
summary$APA

## making a bunch of columns numeric
summary$IO_Max <- gsub("%", "", summary$IO_Max) #remove the percent sign
summary[, c(5:15)] <- sapply(summary[, c(5:15)], as.numeric)

## create genoAPA column
summary$genoAPA <- summary$genoAPA <- as.factor(paste(summary$genotype,summary$APA, sep="_"))
summary$genoAPA <- factor(summary$genoAPA, levels = c("WT_untrained" ,"WT_trained"  , "FMR1-KO_untrained","FMR1-KO_trained" ))
summary$genoAPA

head(summary)
summary_long <- melt(summary, id=c("ind","genotype", "APA", "session", "filename", "genoAPA"))
summary_long$value <- as.numeric(summary_long$value)
str(summary_long)
levels(summary_long$genoAPA)
levels(summary_long$genoAPA) <- c("WT_untrained" ,"WT_trained"  , "FMR1-KO_untrained","FMR1-KO_trained" )


### improve indiviual session dataframe ----
str(wtfmr1)
wtfmr1 <- wtfmr1[c(60:61,62,1:45)]  # removes summary columns

##create APA dataframe to add APA to wtfrm1
APA <-  summary[c(1:3)] 
wtfmr1 <- left_join(wtfmr1,APA) 



## create columns for genotype*APA, genotype*APA*session, and genotype*APA*session*IND
wtfmr1$genoAPA <- as.factor(paste(wtfmr1$genotype,wtfmr1$APA, sep="_"))
wtfmr1$genoAPA <- factor(wtfmr1$genoAPA, levels = c("WT_untrained", "FMR1-KO_untrained", "WT_trained", "FMR1-KO_trained"))
wtfmr1$genoAPAsession <- as.factor(paste(wtfmr1$genoAPA, wtfmr1$session, sep="_")) #create genoAPAsession column
wtfmr1$genoAPAsessionInd <- as.factor(paste(wtfmr1$genoAPAsession, wtfmr1$ind, sep="_")) #create genoAPAsessionInd column
head(wtfmr1)
names(wtfmr1)

# reorders dataframe
wtfmr1 <- wtfmr1[c(1:4,49:52,5:48)]  
names(wtfmr1)

### melt the wtfmr1 df to make long for graphics
wtfmr1_long <- melt(wtfmr1, id=c("ind","genotype", "APA", "session", "genoAPA", "genoAPAsession", "genoAPAsessionInd", "filename"))
wtfmr1_long$value <- as.numeric(wtfmr1_long$value)
str(wtfmr1_long)
head(wtfmr1_long)


wtfmr1Num <- wtfmr1
wtfmr1Num$sessionNum <- ifelse(grepl("pretraining", wtfmr1Num$session), "1", 
                                     ifelse(grepl("training1", wtfmr1Num$session), "2",
                                            ifelse(grepl("training2", wtfmr1Num$session), "3",
                                                   ifelse(grepl("training3", wtfmr1Num$session), "4", 
                                                          ifelse(grepl("retention", wtfmr1Num$session), "6", 
                                                                 ifelse(grepl("retest", wtfmr1Num$session), "5", NA))))))
wtfmr1Num$sessionNum <- as.numeric(as.character(wtfmr1Num$sessionNum))
wtfmr1NumSummary <- dplyr::summarise(group_by(wtfmr1Num, genotype, APA, genoAPA, sessionNum), m = mean(Entrances), se = sd(Entrances)/sqrt(length(Entrances)), len = length(Entrances))
wtfmr1NumSummary <- as.data.frame(wtfmr1NumSummary)

#write.csv(wtfrm1, "wtfmr1_forMaddy.csv", row.names = F)
```


### Experimental Design
```{r}
APA$genoAPA <- as.factor(paste(APA$genotype,APA$APA, sep="_"))
summary(APA)
```



###  Avoidance behavior: number of entrances

```{r 1-numentrances}
wtfmr1NumSummary %>%
  ggplot(aes(x= sessionNum, y=m, shape=genotype)) + 
  geom_errorbar(aes(ymin=m-se, ymax=m+se, color=genoAPA), width=.1) +
  geom_line(aes(colour=genoAPA, linetype=genoAPA)) +
  geom_point(size = 2.5, aes(colour=genoAPA, shape=genotype)) +
  scale_color_manual(values = colvalAPA12) +
  scale_y_continuous(name="Number of Entrances",
                     limits = c(0,35)) +
  scale_x_continuous(name = NULL, 
                     breaks = c(1, 2, 3, 4, 5, 6),
                     labels=c("1" = "Hab. ", "2" = "T1", "3" = "T2", 
                              "4" = "T3", "5" = "Retest", "6" = "Reten.")) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  background_grid(major = "y", minor = "y") +
  scale_linetype_manual(values = c(1,2,1,2)) +
  scale_shape_manual(values=c(16, 1))

wtfmr1NumSummary$behvaior <- "Number of Entrances"
numentrances <- wtfmr1NumSummary %>%
  ggplot(aes(x= sessionNum, y=m, shape=genotype)) + 
  geom_errorbar(aes(ymin=m-se, ymax=m+se, color=genoAPA), width=.1) +
  geom_line(aes(colour=genoAPA, linetype=genoAPA)) +
  geom_point(size = 2.5, aes(colour=genoAPA, shape=genotype)) +
  scale_color_manual(values = colvalAPA12) +
  scale_y_continuous(name="Number of Entrances",
                     limits = c(0,35)) +
  scale_x_continuous(name = NULL, 
                     breaks = c(1, 2, 3, 4, 5, 6),
                     labels=c("1" = "Hab. ", "2" = "T1", "3" = "T2", 
                              "4" = "T3", "5" = "Retest", "6" = "Reten.")) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  background_grid(major = "y", minor = "y") +
  theme( legend.position="none",
         axis.title.y = element_blank()) +
  scale_linetype_manual(values = c(1,2,1,2)) +
  scale_shape_manual(values=c(16, 1)) +
  facet_wrap(~behvaior)
numentrances



pdf(file="../figures/1-numentrances.pdf", width=2.25, height=2)
plot(numentrances)
dev.off()

```

## Electrophysiology

### IO Max

```{r 2-io_max}
summary_long %>%
  filter(grepl("IO_Max", variable))%>% 
  filter(!grepl("LTP_Baseline_SD", variable))%>% 
  ggplot(aes(x=genoAPA, y=value, color=genoAPA, fill=genoAPA)) +
  geom_violin() +  
  facet_wrap(~variable, scales = "free_y") +
  theme_bw() +
  scale_y_continuous(name=NULL) + 
  scale_x_discrete(name=NULL) + 
  scale_colour_manual(values=colvalAPA12) +
  theme_bw() + 
  theme(panel.grid.minor = element_blank(),
        axis.text.x = element_blank(), 
        axis.ticks = element_blank())  +
  scale_fill_manual(values=colorvalgenoAPA2)

iomax <- summary_long %>%
  filter(grepl("IO_Max", variable))%>% 
  ggplot(aes(x=genoAPA, y=value, color=genoAPA, fill=genoAPA)) +
  geom_violin() +  
  facet_wrap(~variable, scales = "free_y") +
  theme_bw() +
  scale_y_continuous(name=NULL) + 
  scale_x_discrete(name=NULL) + 
  scale_colour_manual(values=colvalAPA12) +
  theme_cowplot(font_size = 8, line_size = 0.1) +
  theme(panel.grid.minor = element_blank(),
        axis.text.x = element_blank(), 
        legend.position="none",
        axis.title = element_blank())  +
  scale_fill_manual(values=colorvalgenoAPA2) 
iomax

pdf(file="../figures/2-iomax.pdf", width=1.25, height=2)
plot(iomax)
dev.off()
```


```{r qpcrdatawrangle, echo=F }
# wrangle the count data
qpcr <- read.csv("../data/02_qpcrdata.csv", header = TRUE, na.strings = "NA", stringsAsFactors = FALSE)
qpcr$sample <- as.factor(qpcr$sample)
qpcr$ind <- as.factor(qpcr$ind)
qpcr$time <- as.factor(qpcr$time)
qpcr$region <- as.factor(qpcr$region)
qpcr$APA <- as.factor(qpcr$APA)
qpcr$strain <- as.factor(qpcr$strain)
qpcr$X <- NULL
qpcr <- dplyr::rename(qpcr, genotype = strain) 
qpcr$genotype <- revalue(qpcr$genotype, c("fmr1" = "FMR1-KO")) 
qpcr$genotype <- revalue(qpcr$genotype, c("wt" = "WT"))
qpcr$APA <- revalue(qpcr$APA, c("control" = "untrained"))
names(qpcr)[7] <- "prkcz"
qpcr$ind <- revalue(qpcr$ind, c("BL AB" = "FMR1 AB"))
qpcr[10:12, 6] = "FMR1-KO"
qpcr$region.genotype <- as.factor(paste(qpcr$region, qpcr$genotype, sep="_"))
qpcr$genoAPA <- as.factor(paste(qpcr$genotype,qpcr$APA, sep="_"))
qpcr$genoAPAregion <- as.factor(paste(qpcr$genoAPA,qpcr$region, sep="_"))
qpcr$genoAPA <- factor(qpcr$genoAPA, levels = c("WT_untrained" ,"WT_trained"  , "FMR1-KO_untrained","FMR1-KO_trained"))
## reorder the dataframe
qpcr <- qpcr[c(1:6,19:21,7:18)]

## calculating gene efficiencies & rename genes ----
dilutions <- read.csv("../data/02_dilutions_CA1CA3.csv", header = TRUE)
head(dilutions)
dilutions <- filter(dilutions, gene != "dlg4", gene != "pkmz.conc", gene != "rpl19.conc", gene != "grim")
dilutions$gene <- revalue(dilutions$gene, c("dlg4.conc" = "dlg4")) 
dilutions$gene <- revalue(dilutions$gene, c("grim.conc" = "grim"))
dilutions$gene <- revalue(dilutions$gene, c("fmr1.conc" = "fmr1"))
dilutions$gene <- revalue(dilutions$gene, c("rRNA18s" = "rRNA18S"))
dilutions$gene <- revalue(dilutions$gene, c("pkmz" = "prkcz"))
dilutions <- droplevels(dilutions)
PrimEff(dilutions) -> eff
```


```{r qpcrCA1only}
## Analyzing just the  CA1 data ----
CA1only <- dplyr::filter(qpcr, APA != "home", region == "CA1") %>%
  droplevels()
CA1only <- CA1only[1:12]

dd_CA1only <- cq2counts(data=CA1only, genecols=c(10:12), condcols=c(1:9), effic=eff)

naive_dd_CA1only <- mcmc.qpcr(
  data=dd_CA1only,
  fixed="APA",
  pr=T,pl=T, singular.ok=TRUE)
diagnostic.mcmc(model=naive_dd_CA1only, col="grey50", cex=0.8)

dd_CA1only %>%  
  ggplot(aes(x=genoAPA, y=count)) + 
  geom_violin(aes(fill=genoAPA, color=genoAPA)) +  scale_y_log10(name="log10(gene counts)") +
  facet_wrap(~gene, scales = "free_y") +
  scale_x_discrete(name="") +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  theme(axis.text.x = element_blank(), 
        strip.text = element_text(face = "italic")) + 
  scale_color_manual(values = colvalAPA12) +
  scale_fill_manual(values = colorvalgenoAPA2) 


CA1genes <- dd_CA1only %>%  
  ggplot(aes(x=genoAPA, y=count)) + 
  geom_violin(aes(fill=genoAPA, color=genoAPA)) +  scale_y_log10(name="gene counts") +
  facet_wrap(~gene, scales = "free_y") +
  scale_x_discrete(name="") +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  theme(axis.text.x = element_blank(), 
        strip.text = element_text(face = "italic"),
        legend.position = "none") + 
  scale_color_manual(values = colvalAPA12) +
  scale_fill_manual(values = colorvalgenoAPA2) 
CA1genes

pdf(file="../figures/3-CA1genes.pdf", width=4.25, height=2)
plot(CA1genes)
dev.off()

```

## Integration across levels

```{r integrationdatawrangle, echo = F}
## integrative analysis ----

########## gather three dataset and exampine ----
behavior <- wtfmr1
behavior <- melt(wtfmr1, id=c("ind","genotype", "APA", "session", "genoAPA", "genoAPAsession", "genoAPAsessionInd", "filename"))
behavior <- filter(behavior, grepl("Entrances", variable))
behavior$measure <- as.factor(paste(behavior$session, behavior$variable, sep="_"))
behavior <- dplyr::select(behavior, ind, genotype, APA, genoAPA, measure, value)
head(behavior)

phy <- summary
phy <- melt(summary, id=c("ind","genotype", "APA", "session", "genoAPA", "filename"))
head(phy)
phy <- dplyr::rename(phy, measure = variable)
phy <- dplyr::select(phy, ind, genotype, APA,genoAPA, measure, value)
head(phy)

qpcrcounts <- as.data.frame(dd_CA1only)
qpcrcounts <- qpcrcounts %>%
  dplyr::rename(measure = gene, value=count) %>%
  dplyr::select(ind, genotype, APA, genoAPA, measure, value)
qpcrcounts <- dcast(qpcrcounts, ind + genotype + genoAPA + APA ~ measure, fun=mean)
qpcrcountslong <- melt(qpcrcounts, id=c("ind","genotype","genoAPA", "APA"))
qpcrcountslong <- dplyr::rename(qpcrcountslong, measure = variable)
head(qpcrcountslong)

############# merge all and widen!! -----

alllong <- dplyr::bind_rows(behavior, phy, qpcrcountslong)
alllong$ind <- as.factor(alllong$ind)
alllong$measure <- as.factor(alllong$measure)
alllong$APA <- as.factor(alllong$APA)
alllong$genoAPA <- factor(alllong$genoAPA, levels = c("WT_untrained", "WT_trained", "FMR1-KO_untrained" , "FMR1-KO_trained"))

all <- dplyr::bind_rows(behavior, phy, qpcrcountslong)
all$ind <- as.factor(all$ind)
all$measure <- as.factor(all$measure)
all$APA <- as.factor(all$APA)
all <- dplyr::select(all, ind, genotype, APA, genoAPA, measure, value)
all <- dcast(all, ind + genotype + APA + genoAPA ~ measure, value.var= "value")
all$genoAPA <- factor(all$genoAPA, levels = c("WT_untrained", "WT_trained", "FMR1-KO_untrained" , "FMR1-KO_trained"))

```

```{r integrationfigures}

all$correlation1 <- "Entrances vs IO_Max"
correlation <- all %>%
  filter(APA == "trained") %>%
  ggplot(aes(x=training3_Entrances, y=IO_Max, color=genoAPA)) +
  geom_point(size = 1.5, aes(colour=genoAPA, shape=genotype)) +
  geom_smooth(method = lm, alpha = 0.2, aes(linetype=genoAPA)) +
  theme_cowplot(font_size = 8, line_size = 0.1) +
  #theme(legend.position="none") +
  scale_color_manual(values = c("red", "red")) +
  scale_linetype_manual(values = c(1,2)) +
  scale_shape_manual(values=c(16, 1)) +
  facet_wrap(~correlation1)
correlation

pdf(file="../figures/3-correlation1.pdf", width=3.25, height=2)
plot(correlation)
dev.off()

all$correlation2 <- "Entrances vs rpl19"

correlation <- all %>%
  filter(APA == "trained") %>%
  ggplot(aes(x=training3_Entrances, y=log10(rpl19), color=genoAPA)) +
  geom_point(size = 1.5, aes(colour=genoAPA, shape=genotype)) +
  geom_smooth(method = lm, alpha = 0.2, aes(linetype=genoAPA)) +
  theme_cowplot(font_size = 8, line_size = 0.1) +
  theme(legend.position="none",
        axis.title = element_blank()) +
  scale_color_manual(values = c("red", "red")) +
  scale_linetype_manual(values = c(1,2)) +
  scale_shape_manual(values=c(16, 1)) +
  facet_wrap(~correlation2)
correlation

pdf(file="../figures/3-correlation2.pdf", width=3, height=2)
plot(correlation)
dev.off()
```

```{r correlationmatrix}
## then create a matrix of just gene expression data with NAs ommited
allmatrix <- all # prepare for matrix
allmatrix$genoAPAind <- as.factor(paste(allmatrix$genoAPA,allmatrix$ind, sep="_"))
rownames(allmatrix) <- allmatrix$genoAPAind  # set $genoAPAsession as rownames
names(allmatrix)
allmatrix <- allmatrix[-c(1:4,7:12,14:16,18:21,25:27)]  ## remove non-numeric columns
allmatrix <- log10(allmatrix + 1) #log transform all data
allmatrix <- as.matrix(allmatrix)
str(allmatrix)
allmatrix <- na.omit(allmatrix)
allmatrix_cor <- round(cor(allmatrix),2)  # then create a correlation matrix

corrplot(allmatrix_cor, type="lower", tl.col="black", tl.srt=45)


allmatrix <- all %>%
  filter(genotype == "WT", APA == "trained")
allmatrix$genoAPAind <- as.factor(paste(allmatrix$genoAPA,allmatrix$ind, sep="_"))
rownames(allmatrix) <- allmatrix$genoAPAind  # set $genoAPAsession as rownames
names(allmatrix)
allmatrix <- allmatrix[-c(1:4,7:12,14:16,18:21,25:27)]  ## remove non-numeric columns
allmatrix <- log10(allmatrix + 1) #log transform all data
allmatrix <- as.matrix(allmatrix)
str(allmatrix)
allmatrix <- na.omit(allmatrix)
allmatrix_cor <- round(cor(allmatrix),2)  # then create a correlation matrix

corrplot(allmatrix_cor, type="lower", tl.col="black", tl.srt=45)

allmatrix <- all %>%
  filter(genotype == "FMR1-KO", APA == "trained")
allmatrix$genoAPAind <- as.factor(paste(allmatrix$genoAPA,allmatrix$ind, sep="_"))
rownames(allmatrix) <- allmatrix$genoAPAind  # set $genoAPAsession as rownames
names(allmatrix)
allmatrix <- allmatrix[-c(1:4,7:12,14:16,18:21,25:27)]  ## remove non-numeric columns
allmatrix <- log10(allmatrix + 1) #log transform all data
allmatrix <- as.matrix(allmatrix)
str(allmatrix)
allmatrix <- na.omit(allmatrix)
allmatrix_cor <- round(cor(allmatrix),2)  # then create a correlation matrix

corrplot(allmatrix_cor, type="upper", tl.col="black", tl.srt=45)
```






````{r newintegrativefigures}
## with all long
unique(alllong$measure)
levels(alllong$measure)

alllongslim <- alllong %>%
  filter(grepl("IO_Max|pretraining_Entrances|training3_Entrances|grim|prkcz|rpl19", measure)) 

alllongslim$measure <- factor(alllongslim$measure, levels = c("pretraining_Entrances", "training3_Entrances", "IO_Max", "grim", "prkcz", "rpl19"))


behaviorfigure  <- alllongslim %>%
  filter(grepl("training3_Entrances| ", measure)) %>%
  ggplot(aes(x=genoAPA, y=value, color=genoAPA, fill=genoAPA)) +
  geom_violin() +  
  facet_wrap(~measure, scales = "free_y") +
  theme_bw() +
  scale_y_continuous(name=NULL) + 
  scale_x_discrete(name=NULL) + 
  scale_colour_manual(values=colvalAPA12) +
  theme_cowplot(font_size = 8, line_size = 0.1) +
  theme(panel.grid.minor = element_blank(),
        axis.text.x = element_blank(), 
        legend.position="none",
        axis.title = element_blank())  +
  scale_fill_manual(values=colorvalgenoAPA2) 
behaviorfigure

pdf(file="../figures/4-behaviorfigure.pdf", width=1.25, height=2)
plot(behaviorfigure)
dev.off()

qpcrfigure <- alllongslim %>%
  filter(grepl("grim|prkcz|rpl19| ", measure)) %>%
  ggplot(aes(x=genoAPA, y=log10(value), color=genoAPA, fill=genoAPA)) +
  geom_violin() +  
  facet_wrap(~measure, scales = "free_y") +
  theme_bw() +
  scale_y_continuous(name=NULL) + 
  scale_x_discrete(name=NULL) + 
  scale_colour_manual(values=colvalAPA12) +
  theme_cowplot(font_size = 8, line_size = 0.1) +
  theme(panel.grid.minor = element_blank(),
        axis.text.x = element_blank(),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        axis.title = element_blank())  +
  scale_fill_manual(values=colorvalgenoAPA2)  
qpcrfigure


pdf(file="../figures/4-qpcrfigure.pdf", width=3.5, height=2)
plot(qpcrfigure)
dev.off()


```

## Statistics

```{r stats}
aov <- aov(Entrances ~ genoAPA, data=wtfmr1)
summary(aov) 
TukeyHSD(aov, which = "genoAPA") 

aov <- aov(grim ~ genoAPA, data=all)
summary(aov) 
TukeyHSD(aov, which = "genoAPA") 

aov <- aov(prkcz ~ genoAPA, data=all)
summary(aov) 
TukeyHSD(aov, which = "genoAPA") 

aov <- aov(rpl19 ~ genoAPA, data=all)
summary(aov) 
TukeyHSD(aov, which = "genoAPA") 

aov <- aov(IO_Max ~ genoAPA, data=all)
summary(aov) 
TukeyHSD(aov, which = "genoAPA") 
```


